🐛 BUG: 'grep' is not recognized as an internal or external command

### What version of `Wrangler` are you using?

0.0.29? wrangler@alpha

### What operating system are you using?

Windows

### Describe the Bug

When calling `wrangler pages publish` the error 
```
'grep' is not recognized as an internal or external command,
operable program or batch file.

✘ [ERROR] ENOENT: no such file or directory, scandir 'C:\Users\CEbbi\Projects\Website\functions'
``` is thrown. Since this is run in PowerShell I don't have to grep as a callable binary therefore it seems to be erroring. Is this a required dependency I was not aware of?

--- a/.changeset/silly-squids-attend.md
+++ b/.changeset/silly-squids-attend.md
@@ -1,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+fix: Ensure getting git branch doesn't fail on Windows

--- a/packages/wrangler/src/pages.tsx
+++ b/packages/wrangler/src/pages.tsx
@@ -979,9 +979,8 @@ const createDeployment: CommandModule<
         );
 
         if (!branch) {
-          branch = execSync(`git branch | grep "* "`)
+          branch = execSync(`git rev-parse --abbrev-ref HEAD`)
             .toString()
-            .replace("* ", "")
             .trim();
         }
 

--- a/packages/wrangler/src/pages.tsx
+++ b/packages/wrangler/src/pages.tsx
@@ -925,10 +925,7 @@ const createDeployment: CommandModule<
             "Enter the production branch name:",
             "text",
             isGitDir
-              ? execSync(`git branch | grep "* "`)
-                  .toString()
-                  .replace("* ", "")
-                  .trim()
+              ? execSync(`git rev-parse --abbrev-ref HEAD`).toString().trim()
               : "production"
           );
 
@@ -1681,9 +1678,8 @@ export const pages: BuilderCallback<unknown, unknown> = (yargs) => {
                 "Enter the production branch name:",
                 "text",
                 isGitDir
-                  ? execSync(`git branch | grep "* "`)
+                  ? execSync(`git rev-parse --abbrev-ref HEAD`)
                       .toString()
-                      .replace("* ", "")
                       .trim()
                   : "production"
               );

